examples/pipe_closed.c

```diff
@@ -26,7 +26,8 @@ int main(int argc, char **argv)
         } while (res && errno == EINTR);
     } else if (world_rank == 1) {
         test_assert(MIMPI_Recv(&number, 1, 0, tag) == MIMPI_ERROR_REMOTE_FINISHED);
-        test_assert(MIMPI_Send(&number, 1, 0, tag) == MIMPI_ERROR_REMOTE_FINISHED);
+        MIMPI_Retcode const ret = MIMPI_Send(&number, 1, 0, tag);
+        test_assert(ret == MIMPI_ERROR_REMOTE_FINISHED || ret == MIMPI_SUCCESS);
```

Comment:
This now compiles with warnings, because of how test_assert is expanded. Inner expression should be embraced with () to prevent this.